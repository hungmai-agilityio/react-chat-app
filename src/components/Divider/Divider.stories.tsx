import { Meta, StoryObj } from '@storybook/react';

// Components
import Divider from '@/components/Divider';
import Chip from '@/components/Chip';

const meta: Meta<typeof Divider> = {
  component: Divider,
  tags: ['autodocs'],
  argTypes: {
    children: {
      description: 'Render dynamic react node children'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Divider>;

export const Primary: Story = {
  args: {}
};

export const WithChildrenComponent: Story = {
  args: {
    children: (
      <Chip>
        <p>June, 04 2024</p>
      </Chip>
    )
  }
};

export const WithChildrenText: Story = {
  args: {
    children: 'Test'
  }
};
