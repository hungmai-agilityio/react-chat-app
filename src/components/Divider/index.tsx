import { ReactNode } from 'react';

interface DividerProps {
  children?: ReactNode;
}

const Divider = ({ children }: DividerProps) => {
  const styleDefault = 'border-b border-gray-300 w-full';

  return (
    <div className="flex items-center">
      <div className={styleDefault} />
      {children && <div className="px-2">{children}</div>}
      <div className={styleDefault} />
    </div>
  );
};

export default Divider;
