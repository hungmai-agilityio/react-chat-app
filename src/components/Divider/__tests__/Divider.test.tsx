import { render, screen } from '@testing-library/react';

// Components
import Divider from '@/components/Divider';

describe('Divider', () => {
  it('render Divider children', () => {
    render(<Divider children="Test" />);

    const childrenProp = screen.getByText('Test');
    expect(childrenProp).toBeInTheDocument();
  });
});

it('should render match to snapshot', () => {
  const { container } = render(<Divider children="Test" />);

  expect(container).toMatchSnapshot();
});
