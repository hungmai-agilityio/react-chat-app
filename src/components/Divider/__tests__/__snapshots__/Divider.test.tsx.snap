// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render match to snapshot 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <div
      class="border-b border-gray-300 w-full"
    />
    <div
      class="px-2"
    >
      Test
    </div>
    <div
      class="border-b border-gray-300 w-full"
    />
  </div>
</div>
`;
